azurerm/resource_arm_postgresql_server.go

```diff
@@ -0,0 +1,306 @@
+package azurerm
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+
+	"github.com/Azure/azure-sdk-for-go/arm/postgresql"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/jen20/riviera/azure"
+)
+
+func resourceArmPostgreSQLServer() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceArmPostgreSQLServerCreate,
+		Read:   resourceArmPostgreSQLServerRead,
+		Update: resourceArmPostgreSQLServerUpdate,
+		Delete: resourceArmPostgreSQLServerDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"location": locationSchema(),
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+
+			"sku": {
+				Type:     schema.TypeSet,
+				Required: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"name": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								"PGSQLB50",
+								"PGSQLB100",
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+						},
+
+						"capacity": {
+							Type:     schema.TypeInt,
+							Required: true,
+							ValidateFunc: validateIntInSlice([]int{
+								50,
+								100,
+							}),
+						},
+
+						"tier": {
+							Type:     schema.TypeString,
+							Required: true,
+							ValidateFunc: validation.StringInSlice([]string{
+								string(postgresql.Basic),
+							}, true),
+							DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
```

Comment:
🤔 Doesn't this cause issues to default hash function, as the field is part of `Set`?

Reply:
Nope - it just allows the user to input `basic` rather than `Basic` - providing they do that consistently there shouldn't be an issue here (regardless, it'd be nice to switch these to Lists rather than Sets in hindsight)